### ISSUE TITLE
Production build has many side effects and application is running totally wrong

### ISSUE THREAD

**Issue by jimmykane** [angular/angular#19171]
## I'm submitting a...
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
I have created an app for parsing gpx files and showing info on map etc. 
All works fine when running for developing env but when I build for production all stats maps get messed up. 

## Expected behavior
The application should run the same fro production and dev

## Minimal reproduction of the problem with instructions
1. Download my opensource app: https://github.com/jimmykane/quantified-self
2. Run it for dev
3. Upload a tcx or gpx file eg fromhttps://raw.githubusercontent.com/jimmykane/quantified-self/master/samples/track_examples/7628882846070661.tcx
4. Test the app, you should be getting a map with all the info needed, such as speed cadence places etc. 
5. Build for production. 
6. Run the app
7. Everything is messed up. Maps show wrong places and many more unexpected behaviour

## What is the motivation / use case for changing the behavior?
I was building the app now for 3 months and I am suprised it has a different behaviour for production

## Environment

<pre><code>
Angular version: 4.3.6

Browser:
- [ x] Chrome (desktop) version XX
- [ x] Chrome (Android) version XX
- [ x] Chrome (iOS) version XX
- [ x] Firefox version XX
- [ x] Safari (desktop) version XX
- [ x] Safari (iOS) version XX
- [ x] IE version XX
- [ x] Edge version XX

**Comment by Toxicable** [comment]
In v5 AOT is likely going to be the default.
If you're reporting a bug please clearly explain what the bug is
>Everything is messed up

Is not a reasonable explanation of a bug

**Comment by jimmykane** [comment]
@Toxicable Coming up with more info

**Comment by jimmykane** [comment]
My app shows on Google Maps a tcx file for example.  (https://raw.githubusercontent.com/jimmykane/quantified-self/master/samples/track_examples/7628882846070661.tcx)

This is how it looks on develop more:

<img width="1439" alt="screen shot 2017-09-13 at 11 29 52" src="https://user-images.githubusercontent.com/1268723/30370154-06ec9402-9877-11e7-92d7-df5cb3498fe8.png">


And this is how it looks on production: 

<img width="1436" alt="screen shot 2017-09-13 at 11 31 33" src="https://user-images.githubusercontent.com/1268723/30370205-2b6b499a-9877-11e7-91fa-e41c32d61042.png">


This looks like an arithmetic problem. 

Actually the truth is that I do not really know what can be going wrong and this is why I am here...

@Toxicable does this make any sense?

**Comment by jimmykane** [comment]
@Toxicable I really dont know what can be causing this so I do not know how to express my issue here. A bit of help would be more than appreciated

**Comment by Toxicable** [comment]
Sure that makes sense. But I don't see what the issue is with Angular here.
There are far too many variables that come in to play, it's impossible to tell if it's Angulars fault from what you've shown.
Take some time to debug this issue for yourself and figure out what is causing it, if you can make a minimal reproduction showing a bug with Angular then post that here, otherwise this is not the right place to ask for help.

**Comment by jimmykane** [comment]
@Toxicable The part that I do not understand and there is where I am stuck is that what can be the difference when compiling for dev and when compiling for prod. 

The app has grown big and I did not expect this to happen. 

So I need an indication where to start please.

**Comment by jimmykane** [comment]
Also this should not be an expected behaviour. 
What I mean with that, is that compiling for Prod, should not take more testing since it should be a blind copy from dev.

**Comment by Toxicable** [comment]
When you invoke the command  `ng build --prod` you enable AOT compiling, this is not the same as `ng build` where it runs in JIT.
see here https://angular.io/guide/aot-compiler and https://angular.io/guide/metadata for more info

**Comment by jimmykane** [comment]
@Toxicable is there a way to disable uglify for prod build? I also suspect that can be a case.

**Comment by jimmykane** [comment]
@Toxicable I run the dev build with `ng build --aot=true`. 
That means that AOT is enabled for the dev build correct? 
If that is correct then AOT is not the issue here since with this build all looks fine. 

I will investigate more and come back.

**Comment by jimmykane** [comment]
Back on this. @Toxicable  you are right. 

What I did I used atom for dev built. All was fine. 
However uglifying the source had an issue with the produced code as the IDE was configured for ES6, and the built process worked just fine, no errors. 
To be more precise this part of code https://github.com/jimmykane/quantified-self/commit/78d67b4373bb8dbe1661b527568d83c9c21d9f42
would get compiled successfully. 

Btw thanks a lot also about the mention in the other issue and there you are correct as well. However as you see I am coming from an ES6 project and this created the confusion. 

Thanks for your time and knowledge.
